File: tvr-cmdline/src/main/java/com/willcurrie/DecodedWriter.java
package com.willcurrie;

import java.io.PrintStream;
import java.util.List;

public class DecodedWriter {
    private PrintStream out;

    public DecodedWriter(PrintStream out) {
        this.out = out;
    }

    public void write(List<DecodedData> decoded, String indent) {
        for (DecodedData d : decoded) {
            out.print(indent);
            out.print("[");
            out.print(d.getRawData());
            out.print("] ");
            out.print(d.getDecodedData());
            out.print("\n");
            write(d.getChildren(), indent + "  ");
        }
    }
}

File: src/main/java/io/github/binaryfoo/cmdline/DecodedWriter.kt
package io.github.binaryfoo.cmdline

import io.github.binaryfoo.DecodedData

import java.io.PrintStream

public class DecodedWriter(private val out: PrintStream) {

    public fun write(decoded: List<DecodedData>, indent: String) {
        for (d in decoded) {
            out.print(indent)
            if (d.rawData.length() > 0) {
                out.print("[")
                out.print(d.rawData)
                out.print("] ")
            }
            out.print(d.getDecodedData())
            out.print("\n")
            write(d.children, indent + "  ")
        }
    }
}
